src/test/groovy/graphql/GraphQLTest.groovy

```diff
@@ -1015,4 +1015,132 @@ many lines''']
 
     }
 
+    def "default argument values are respected when variable is not provided"() {
+        given:
+        def spec = """type Query {
+            sayHello(name: String = "amigo"): String
+        }"""
+        DataFetcher df = { dfe ->
+            return dfe.getArgument("name")
+        } as DataFetcher
```

Comment:
I think in groovy if you make the variable be a type then there is no need for `as Foo`

`DataFetcher dfVar = { env -> "x" }`